src/modules/credentials/services/CredentialService.ts

```diff
@@ -403,12 +429,15 @@ export class CredentialService {
     const { comment } = options
     const credentialRequest = new RequestCredentialMessage({
       comment,
-      requestAttachments: [attachment],
+      requestAttachments: [requestAttachment],
+      attachments: credentialRecord.offerMessage?.attachments,
     })
+
     credentialRequest.setThread({ threadId: credentialRecord.tags.threadId })
 
     credentialRecord.metadata.requestMetadata = credReqMetadata
     credentialRecord.requestMessage = credentialRequest
+    credentialRecord.attachments = credentialRecord.offerMessage?.attachments
```

Comment:
You're not filtering the attachments here

Reply:
The only way to get an `attachment` in the offerMessage is if it were a `linkedAttachment` before. So it seems a bit redundant.

Reply:
That's not true. You could have received a message with any attachments.

Reply:
Or if that's true, then it should be called `linkedAttachments`, not `attachments`

Reply:
Maybe we're having a misunderstanding here. I'm not saying it should be `LinkedAttachment` class instance in this case. I'm saying that `Attachment` could be meaning a linked attachment, or a non-linked attachment

Reply:
So, the attribute in the credentialRecord should be called `linkedAttachments` and contains instances of the `Attachment` class? I am fine either way, I just chose this name to stay consistent with the classname.

Reply:
I think so. I agree it's consistent with the class name, but it is leaving an important part of the context out. Namely that the `Attachments` in that array are only attachments used for linked attachments, not general attachments